capstone-app/src/app/modules/history/wiki-media/wiki-media.component.ts

```diff
@@ -63,19 +64,16 @@ export class WikiMediaComponent implements OnChanges, OnInit {
     let isOldRequest = false;
     this.wikiService.search(queryString, 'en', 'History')
       .subscribe((result: WikiServiceResult) => {
-        this.error = "";
+        this.error = '';
         if (result.history && currentQuery === this.cityName) {
           this.body = result.history;
           this.history = result.history;
           this.urls = result.furtherReading;
           this.title = result.title;
           this.loading = false;
           this.langlinks = result.langlinks;
-          const language = this.langlinks.get('ca');
-          this.prevQuery = language.get('searchQuery');
           this.prevWordForHistory = 'History';
           this.prevPreFix = 'en';
-          this.cd.detectChanges();
```

Comment:
This needed to be removed? Can you explain why so I understand better what you did?

Reply:
If this line was kept in then there was a change detection dom issue with the mat menus for langlinks and further reading. The change I made to change detection was in ngOnInit, I wrapped this.getResults() in an ngZone.run which catches the function as it leaves the Angular zone and puts it back in. Basically. It fixes the change detection.

Reply:
Ah, ok. Thanks for the detailed explanation!